src/notebooklm/cli/helpers.py

```
@@ -96,17 +141,25 @@ async def import_with_retry(
             remaining = max_elapsed - elapsed
             if remaining <= 0:
                 raise
 
             sleep_for = min(delay, max_delay, remaining)
+            # Filter out sources already imported to avoid duplicates on retry
+            try:
+                existing = await client.sources.list(notebook_id)
+                existing_urls = {s.get("url") for s in existing if s.get("url")}
+                pending_sources = [s for s in pending_sources if s.get("url") not in existing_urls]
+            except Exception:
+                pass  # If listing fails, retry with original list
```

Ruff (0.15.7)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)